platform/retarget.cpp

```diff
@@ -350,7 +396,13 @@ long __lseek(int fh, long offset, int whence)
 int _lseek(FILEHANDLE fh, int offset, int whence)
 #endif
 {
-    if (fh < 3) return 0;
+    if (fh < 3)
+        return 0;
+#if defined MBED_CFG_DEBUG_OPTIONS_COVERAGE
+    else if (coverage_report && fh == gcov_fd){
+        return 0;
+    }
+#endif
```

Comment:
Nit: Same deal in the above block about spaces, there should be spaces after the `if ()`.

Ex. `if (blah){` should be `if (blah) {`.